Pronto_ROS2_Node :
  ros__parameters:
    use_sim_time: True
    pose_topic : "pose_est"
    pose_frame_id : "odom"
    twist_topic : "twist_est"
    twist_frame_id : "base_link"
    filter_state_topic : "filter_state"
    publish_filter_state : false
    publish_pose : true
    publish_tf : true
    tf_child_frame_id : "base_pronto"
    republish_sensors : false
    init_sensors : ["ins"]
    active_sensors : [ "scan_matcher","ins"]
    utime_history_span : 600000


    sigma0 :
      vb : 0.5 # m/s
      chi_xy : 3.0 # degrees?
      chi_z : 3.0 # degrees?
      Delta_xy : 0.05 #0.5 # m
      Delta_z : 0.05 # 1   # m
      gyro_bias : 0.0 # deg/s
      accel_bias : 0.0 # m/s^2 # 0.1 worked during tuning

    x0 :
      velocity : [ 0.0, 0.0, 0.0 ]
      angular_velocity : [ 0.0, 0.0, 0.0 ]
      position : [ 0.0, 0.0, 0.0 ]
      rpy : [ 0.0, 0.0, 0.0 ]

    init_message :
      channel : "filter_initializer"

    ins:
      topic : "/IMU_Broadcaster/imu"
      utime_offset : 0 # units 1e-6 seconds
      downsample_factor : 1 # process 1 out of every this many. for 333Hz use 1. for 1kHz use 3
      roll_forward_on_receive : true
      publish_head_on_message : true

      # don't integrate acceleration, rely on some other source for velocity/position
      ignore_accel : false

      # not clear what value to set for the omega, if it is rad/s it seems too much
      q_gyro : 0.01 # deg / s
      q_accel : 0.5 # m / s^2, taken from any_common/any_description/urdf/sensors/imu/xsens_mti/xsens_mti.urdf.xacro
      q_gyro_bias : 0.0 # deg / sec^2 | 0.001 suggested by charlie, but not deemed to be necessary
      q_accel_bias : 0.0 # m / sec^2 / sec | 0.001 suggested by charlie and worked well on logs

      # number of samples for the bias estimation
      num_to_init : 100

      accel_bias_initial : [0.0, 0.0, 0.0]
      accel_bias_recalc_at_start : true
      accel_bias_update_online : false

      gyro_bias_initial : [0.0, 0.0, 0.0] # rad/s
      gyro_bias_recalc_at_start : true
      gyro_bias_update_online : false

      frame : "base_link"
      base_link_name: "base_link"
      # REMEMBER TO CHANGE THIS IF YOU CHANGE THE IMU FREQUENCY or downsample factor
      timestep_dt: 0.002 #

      # attempt to detect drastically incorrect initial gyro biases estimates:
      max_initial_gyro_bias : 0.015 # deg/sec, biases above this will be set to zero

    scan_matcher:
        mode: "position"
        r_pxy: 0.001
        r_pz: 0.001
        r_yaw: 0.001
        topic: "/odom_rf2o"
        roll_forward_on_receive : false
        publish_head_on_message : false

    qualysis_mt:
        robot_name: "Omnicar"

        r_xyz: 0.001
        r_chi: 0.1
        mode: 0
        topic: "/rigid_bodies"
        roll_forward_on_receive : false
        publish_head_on_message : false
